oe_translation.module

```diff
@@ -2,7 +2,199 @@
+/**
+ * Implements hook_form_FORM_ID_alter() for the Local Task item edit form.
+ */
+function oe_translation_form_tmgmt_local_task_item_edit_form_alter(&$form, FormStateInterface $form_state) {
+  /** @var \Drupal\tmgmt_local\LocalTaskItemInterface $task_item */
+  $task_item = $form_state->getBuildInfo()['callback_object']->getEntity();
+  $job = $task_item->getJobItem()->getJob();
+  try {
+    $translator_plugin = \Drupal::service('plugin.manager.tmgmt.translator')->createInstance($job->getTranslator()->getPluginId());
+    if ($translator_plugin instanceof AlterableTranslatorInterface) {
+      $translator_plugin->localTaskItemFormAlter($form, $form_state);
+    }
+  }
+  catch (PluginNotFoundException $exception) {
+    return;
+  }
+}
+
+/**
+ * Implements hook_form_alter() for the Job item edit form.
+ *
+ * This is the form used to review the translation.
+ */
+function oe_translation_form_tmgmt_job_item_edit_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  /** @var \Drupal\tmgmt\JobItemInterface $job_item */
+  $job_item = $form_state->getBuildInfo()['callback_object']->getEntity();
+  $job = $job_item->getJob();
+  try {
+    $translator_plugin = \Drupal::service('plugin.manager.tmgmt.translator')->createInstance($job->getTranslator()->getPluginId());
+    if ($translator_plugin instanceof AlterableTranslatorInterface) {
+      $translator_plugin->jobItemFormAlter($form, $form_state);
+    }
+  }
+  catch (PluginNotFoundException $exception) {
+    return;
+  }
+}
+
+/**
+ * Implements hook_entity_type_alter().
+ */
+function oe_translation_entity_type_alter(array &$entity_types) {
+  foreach ($entity_types as $entity_type_id => $entity_type) {
+    if (!$entity_type->hasHandlerClass('moderation')) {
+      continue;
+    }
+
+    $entity_type->setHandlerClass('moderation', TranslationModerationHandler::class);
+  }
+}
+
+/**
+ * Implements hook_tmgmt_translatable_fields_alter().
+ */
+function oe_translation_tmgmt_translatable_fields_alter(ContentEntityInterface $entity, &$translatable_fields) {
+  if (isset($translatable_fields['moderation_state'])) {
+    unset($translatable_fields['moderation_state']);
+  }
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_access() for the Local Task Item.
+ */
+function oe_translation_tmgmt_local_task_item_access(EntityInterface $entity, $operation, AccountInterface $account) {
+  $access = AccessResult::neutral();
+
+  /** @var \Drupal\tmgmt\TranslatorInterface[] $translators */
+  $translators = \Drupal::entityTypeManager()->getStorage('tmgmt_translator')->loadMultiple();
+  foreach ($translators as $translator) {
+    $plugin_id = $translator->getPluginId();
+    try {
+      $translator_plugin = \Drupal::service('plugin.manager.tmgmt.translator')->createInstance($plugin_id);
+      if ($translator_plugin instanceof LocalTranslatorInterface) {
+        $access = $access->orIf($translator_plugin->localTaskItemAccess($entity, $operation, $account))->addCacheableDependency($translator);
```

Comment:
There is one thing I don't understand here: I assumend local_task_items already have a translator assigned to them, so why check other translators to define the access?

Reply:
Here we are allowing any local translators to influence the access to the local task items. It's not about having translators assigned to tasks. We are abstracting away from the concept of "assignment" when we deal with local translations. If you saw, the process of translating content doesn't see things like jobs and assignments. It tries to mirror a bit what core does.